models/code_climate.py

```diff
@@ -4,6 +4,28 @@
 
 
 class CodeClimate(QualityModel):
+    # Arbitrary choice of 30 minutes implementation time per line
+    _LINE_IMPLEMENTATiON_TIME = 30
+    # Code Climate reports 45 minutes to fix when 3 over the allowed cognitive complexity
+    _COGNITIVE_COMPLEXITY_REMEDIATION_COST = 45
+    # Too many return statements is reported as 30 minutes to fix
+    _RETURN_STATEMENTS_REMEDIATION_COST = 30
+    # Nested control flow is reported as taking 45 minutes to fix
+    _NESTED_CONTROL_FLOW_REMEDIATION_COST = 60
+    # When argument count is exactly one over the threshhold (5) it takes 35 minutes
+    # to fix It seems to scale with 10 minutes per argument hat is over when it is (6)
+    # then it takes 45 minutes to fix
+    _ARGUMENT_COUNT_REMEDIATION_COST = 35
+    # When the method is 9 lines over 25, Code Climate reports that it takes 1 hour to fix
+    _METHOD_LINES_REMEDIATION_COST = 60
+    # When the file is exactly 1 line over 250, Code Climate reports that it will take 2 hours to fix
+    _FILE_LINES_REMEDIATION_COST = 120
+    # When there is code duplication in 2 places, Code Climate reports that it will take 1 hour to fix
+    # It does, however, appear that they account for the size of the code snippet somehow as they sometimes
+    # report 45, 50 mminutes
+    _IDENTICAL_CODE_REMEDIATION_COST = 60
+    _SIMILAR_CODE_REMEDIATION_COST = 60
```

Comment:
Where are these numbers coming from? Should we consider writing down a source (or reasoning) somewhere, or at the very least, discuss it in the report?

Reply:
The remediation costs are semi-arbitrary and I picked them from just looking at a run of code climate on the testing-data repository. While ours are constant, I am sure Code Climate scales the remediation costs based on the size of the violation.